### ISSUE TITLE
A Small Bug On Mouse Cursor

### ISSUE THREAD

**Issue by KiddoV** [wailsapp/wails#1503]
There is another really really... small bug about the cursor I forgot to mention back then.
When the window in full screen mode (maximize), the mouse cursor should NOT change to ``resize arrow`` when hover the edges of the window.

_Originally posted by @KiddoV in https://github.com/wailsapp/wails/issues/1087#issuecomment-1168646102_

**Comment by leaanthony** [comment]
Can't you normally resize a maximised (not fullscreen) window?

**Comment by leaanthony** [comment]
Depends on #1349

**Comment by leaanthony** [comment]
Bumping this to v2.2. I think we need to rework how we detect window status (fullscreen/maximised/minimised) to make this work properly. Waiting for a callback to Go is not a great solution: I'd like the state pushed to JS as soon as it happens so the test for resizing can be instant.

More info: https://stackoverflow.com/a/47911169

**Comment by glitchedgitz** [comment]
Hi @leaanthony,
Any update on this bug, as v2.2.0 is out?

Indeed it's a small bug but it prevents me from dragging the window. (When the cursor changes on edges in fullscreen)

**Comment by leaanthony** [comment]
Oh sorry no - it slipped from the release. It's an aggressive roadmap. I think we need to pare it down somewhat and this particular bug isn't a high priority. Would be happy to accept a PR if you can create a fix though?

**Comment by glitchedgitz** [comment]
Sure, Can you point me in some direction?

**Comment by leaanthony** [comment]
To fix the original bug, when a cursor hits the edges, we want to first check the window is fullscreen and if so, cancel out the drag operation

**Comment by glitchedgitz** [comment]
Here is a semi-solution.
So it's fixed when the user will use buttons to maximize and minimize (Basically when using functions, it fixed it)
<img width="600px" src=https://user-images.githubusercontent.com/58145476/206886473-5dd44a5b-0861-4952-93ba-779896fa3814.png>

But the common way users will go maximize is by dragging the title and hitting the top to maximize.

The best way would be if we have an event handler so that whenever the window goes maximize we can change `enableResize`.

Or may be using `WindowIsMaximised()` in `v2\internal\frontend\runtime\desktop\main.js`, Like this

> Note: Code needs to be updated as the function returns promise, but just for example.
<img width="600px" src=https://user-images.githubusercontent.com/58145476/206886259-ef1d8a77-eda4-4975-8c4c-3a04a9b7eb9c.png>

**Comment by leaanthony** [comment]
Yeah, it does feel like the state should be available in the frontend without making a call to the backend. And this needs to be per window in the future 😅

**Comment by glitchedgitz** [comment]
Well in that case I think `w` & `f` both need to be connected with the parent struct.
So I can use `f`'s funcitons in `w`. Consider `r`(root) as the parent struct, so I can use `r.f.ExecJS`

**Comment by stffabi** [comment]
You can directly access chromium and do an eval, like in this snippet in window.go

https://github.com/wailsapp/wails/blob/eb2d929824c118c29bb95cc89114bbefe5e0e5c4/v2/internal/frontend/desktop/windows/window.go#L170-L172

**Comment by glitchedgitz** [comment]
Great! This worked when I **maximized**.

But when I put to use the same code in the else statement for **unmaximized** it crashes and shows a ` nil memory address` error.
I tried different things but not working.

**Comment by glitchedgitz** [comment]
I think if we invoke the code after the window appears it should work.

**Comment by glitchedgitz** [comment]
> Yeah, it does feel like the state should be available in the front end without making a call to the back end. And this needs to be per window in the future 😅

I like this Idea, seems like it will solve multiple problems including zoom problems as well.

**Comment by leaanthony** [comment]
Yeah, state sync is a general problem that should be usable in user-space as well as Wails internals...

**Comment by stffabi** [comment]
This bug also applies to Linux.

**Comment by stffabi** [comment]
On macOS the resize cursor are also shown, but on that platform it's possible to resize from maximized.

**Comment by gerbil** [comment]
Any news? I can see the same in v3.

**Comment by oSethoum** [comment]
Still an issue on V3

**Comment by leaanthony** [comment]
I believe this is fixed now? Please reopen if it's still an issue 👍

**Comment by omarcosr** [comment]
I'm having this problem with v3-alpha

**Comment by SivanCola** [comment]
Downstream note: Reasonix has merged a Wails v2 user-space workaround in esengine/DeepSeek-Reasonix#6610 (merge a46fc6f47a00ffffeaee6184c4748cac6cc4ae7d). It uses the native maximized state, clears stale resize-edge and cursor state, and preserves edge resizing for restored and near-full normal windows. Verification covered a maximized cold start, maximize and restore through caption controls and title-bar double-click, top and right edge dragging, and Windows CI. This change will ship with the next Reasonix Desktop release; it is not an upstream Wails fix.
